engine/src/main/java/com/redhat/cloud/notifications/processors/email/aggregators/AdvisorEmailAggregator.java

```diff
@@ -220,4 +243,42 @@ private void countRecommendationsAndIncidents(String ruleIncident) {
             incidentCounter.incrementAndGet();
         }
     }
+
+    /**
+     * Sorts the provided map by the "total_risk" element and limits it to the
+     * maximum number of allowed elements specified in
+     * {@link AdvisorEmailAggregator#MAXIMUM_NUMBER_RETURNED_EVENTS}.
+     * @param unsortedMap the map to be sorted and limited.
+     * @return a new map instance that
+     */
+    public LinkedHashMap<String, Map<String, Object>> sortAndLimitMapByRisk(final Map<String, Map<String, Object>> unsortedMap) {
+        // Get all the entries from the map to be able to sort them.
+        final List<Map.Entry<String, Map<String, Object>>> sortedEntries = new ArrayList<>(unsortedMap.entrySet());
+        sortedEntries.sort((o1, o2) -> {
+            final Integer o1TotalRisk = Integer.parseInt((String) o1.getValue().get(TOTAL_RISK));
+            final Integer o2TotalRisk = Integer.parseInt((String) o2.getValue().get(TOTAL_RISK));
+
+            // The order is reversed because we want the list to be in
+            // descending order, since we want the elements sorted from the
+            // highest risk to the lowest one.
+            return o2TotalRisk.compareTo(o1TotalRisk);
+        });
+
+        // Create the resulting map we are about to return. Make it a linked
+        // hash map to preserve the insertion order.
+        final LinkedHashMap<String, Map<String, Object>> resultMap = new LinkedHashMap<>(MAXIMUM_NUMBER_RETURNED_EVENTS);
+
+        // Populate the new map.
+        for (final Map.Entry<String, Map<String, Object>> entry : sortedEntries) {
+            if (resultMap.size() > MAXIMUM_NUMBER_RETURNED_EVENTS - 1) {
+                break;
```

Comment:
Is the Advisor team aware that some data, which may be important (high risk), will be dropped?

Reply:
Let me forward this question to them, I will answer as soon as they reply.

Reply:
They just confirmed that this behavior is the desired one :man_dancing:

Reply:
Thanks for asking!